### PR TITLE
Feature/terra3 61 external vpc

### PR DESCRIPTION
**Pull request by Kaushikkatariya**
Please review PR.
By default, it will use existing VPC concept.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -259,3 +259,45 @@ variable "s3_solution_bucket_policy" {
...
 }
+
+variable "use_an_existing_vpc" {
+  description = "Enables/disables an AWS Application Load Balancer."
+  type        = bool
+  default     = false
+}
+
+variable "external_vpc_id" {
+  type        = any
```
Comment (w13k) [inline variables.tf]: "any" should only be used in specific cases. Here something else can be clearly defined.

File: variables.tf [hunk 2]
```diff
@@ -259,3 +259,45 @@ variable "s3_solution_bucket_policy" {
...
+}
+
+variable "external_private_subnets" {
+  type        = list(string)
+  description = "vpc id of existing vpc."
+  default     = []
+}
+
+variable "external_vpc_private_route_table_ids" {
+  type        = any
```
Comment (w13k) [inline variables.tf]: "any" should only be used in specific cases. Here something else can be clearly defined.

File: variables.tf [hunk 3]
```diff
@@ -259,3 +259,45 @@ variable "s3_solution_bucket_policy" {
...
+  description = "vpc id of existing vpc."
+  default     = []
+}
+
+variable "external_db_subnet_group_name" {
+  type        = any
```
Comment (w13k) [inline variables.tf]: "any" should only be used in specific cases. Here something else can be clearly defined.

File: variables.tf [hunk 4]
```diff
@@ -259,3 +259,45 @@ variable "s3_solution_bucket_policy" {
...
+  description = "vpc id of existing vpc."
+  default     = ""
+}
+
+variable "external_elasticache_subnet_ids" {
+  type        = any
```
Comment (w13k) [inline variables.tf]: "any" should only be used in specific cases. Here something else can be clearly defined.

File: variables.tf [hunk 5]
```diff
@@ -259,3 +259,45 @@ variable "s3_solution_bucket_policy" {
...
+
+variable "external_vpc_id" {
+  type        = any
+  description = "vpc id of existing vpc."
+  default     = ""
+}
+
+variable "external_public_subnets" {
+  type        = list(string)
+  description = "vpc id of existing vpc."
```
Comment (w13k) [inline variables.tf]: Please update all descriptions below, as they don't fit.

File: main.tf [hunk 6]
```diff
@@ -14,12 +14,26 @@ locals {
   domain_name = length(module.dns_and_certificates) == 0 ? "" : module.dns_and_certificates[0].domain_name
 }
 
+
+locals {
+  create_vpc = var.use_an_existing_vpc == false ? true : false
```
Comment (w13k) [inline main.tf]: Please simplify. Actually "create_vpc" is not needed to be created.

**Review by w13k — CHANGES_REQUESTED** [review]

File: main.tf (hunk 6 above)
Comment (Kaushikkatariya) [reply main.tf]: done

**Review by w13k — APPROVED** [review]

**Comment by w13k** [comment]
This PR is included in version 0.13.0 :tada:

**Pull request merged**